File: app/src/main/java/me/devsaki/hentoid/activities/sources/ImhentaiActivity.java
package me.devsaki.hentoid.activities.sources;

import me.devsaki.hentoid.enums.Site;

public class ImhentaiActivity extends BaseWebActivity {

    private static final String DOMAIN_FILTER = "imhentai.xxx";
    private static final String[] GALLERY_FILTER = {"//imhentai.xxx/gallery/"};
    private static final String[] REMOVABLE_ELEMENTS = {".bblocktop", ".er_container", "#slider"};

    Site getStartSite() {
        return Site.IMHENTAI;
    }

    @Override
    protected CustomWebViewClient getWebClient() {
        CustomWebViewClient client = new CustomWebViewClient(getStartSite(), GALLERY_FILTER, this);
        client.restrictTo(DOMAIN_FILTER);
        client.addRemovableElements(REMOVABLE_ELEMENTS);
        return client;
    }
}

File: app/src/main/java/me/devsaki/hentoid/activities/sources/ImhentaiActivity.kt
package me.devsaki.hentoid.activities.sources

import me.devsaki.hentoid.enums.Site

class ImhentaiActivity : BaseWebActivity() {
    companion object {
        private const val DOMAIN_FILTER = "imhentai.xxx"
        private val GALLERY_FILTER = arrayOf("//imhentai.xxx/gallery/")
        private val REMOVABLE_ELEMENTS = arrayOf(".bblocktop", ".er_container", "#slider")
    }

    override fun getStartSite(): Site {
        return Site.IMHENTAI
    }

    override fun createWebClient(): CustomWebViewClient {
        val client = CustomWebViewClient(getStartSite(), GALLERY_FILTER, this)
        client.restrictTo(DOMAIN_FILTER)
        client.addRemovableElements(*REMOVABLE_ELEMENTS)
        client.adBlocker.addToJsUrlWhitelist(DOMAIN_FILTER)
        return client
    }
}